graph/impl/src/main/java/org/jboss/windup/graph/GraphContextImpl.java

```diff
@@ -272,10 +274,19 @@ private TitanGraph initializeTitanGraph()
         // the performance decrease seems to be minimal.
         conf.setProperty("storage.berkeleydb.cache-percentage", 1);
 
+        conf.setProperty("storage.berkeleydb.lock-mode", LockMode.READ_UNCOMMITTED);
+        conf.setProperty("storage.berkeleydb.isolation-level", BerkeleyJEStoreManager.IsolationLevel.READ_UNCOMMITTED);
```

Comment:
I think this will help with the memory consumption as well (the main problem was the number of read locks we had). However is it safe? Don't we ever read from graph in the same time we are writing to it?